classes/CartRule.php

```diff
@@ -578,6 +580,20 @@ public function checkValidity(Context $context, $alreadyInCart = false, $display
             }
         }
 
+        if ($this->reduction_exclude_special) {
+            $products = $context->cart->getProducts();
+            $is_ok = false;
+            foreach ($products as $product) {
+                if (!$product['reduction_applies']) {
+                    $is_ok = true;
+                    break;
+                }
+            }
+            if (!$is_ok) {
+                return (!$display_error) ? false : Tools::displayError('You cannot use this voucher on reduced products');
```

Comment:
I don't think so, unless something changed, it is available to translate from Localization -> Translations -> Error messages, look on other places in this file:
https://github.com/kpodemski/PrestaShop/blob/3b1a3535dd1ddfce860eb27fecfb8e65a22b5c9c/classes/CartRule.php#L514